### PR TITLE
Add openshift-e2e-test-clusterinfra-qe step for cluster in…

### PR DESCRIPTION
**Pull request by sunzhaohua2**
Add step to run cluster infra team's cases in repo https://github.com/openshift/cluster-api-actuator-pkg 
@kuiwang02 @miyadav @huali9 @shellyyang1989 PTAL

### CODE REVIEW

**Comment by sunzhaohua2** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-azure-ipi-disc-fullypriv-amd-mixarch-f9-longduration-cloud

**Comment by sunzhaohua2** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-gcp-ipi-disc-priv-amd-mixarch-f28-longduration-cloud

**Comment by sunzhaohua2** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-aws-ipi-efa-pg-f5

**Comment by sunzhaohua2** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-gcp-ipi-disc-priv-amd-mixarch-f28-longduration-cloud

**Comment by sunzhaohua2** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-aws-ipi-efa-pg-f5

**Comment by sunzhaohua2** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-gcp-ipi-disc-priv-amd-mixarch-f28-longduration-cloud

File: ci-operator/step-registry/openshift-extended/test/clusterinfrastructure/openshift-extended-test-clusterinfrastructure-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+ref:
+  as: openshift-extended-test-clusterinfrastructure
+  from: cluster-api-actuator-pkg-test
```
Comment (kuiwang02) [inline ci-operator/step-registry/openshift-extended/test/clusterinfrastructure/openshift-extended-test-clusterinfrastructure-ref.yaml:3]: @sunzhaohua2 frankly it has nothing with tests-private repo and its image.
so, I do not suggest to put this step here. you'd better to put it your own module.

and also @jianzhangbjz @Xia-Zhao-rh and me are not owner of this step.
so, please do not put us into reviewer and approver.

File: ci-operator/step-registry/openshift/e2e/test/qe/longrun/clusterinfra/OWNERS [hunk 2]
```diff
@@ -0,0 +1,5 @@
+approvers:
+  - jhou1
+  - jianzhangbjz
+  - kuiwang02
```
Comment (kuiwang02) [inline ci-operator/step-registry/openshift/e2e/test/qe/longrun/clusterinfra/OWNERS:4]: @sunzhaohua2 please remove @jianzhangbjz and me.

File: ci-operator/step-registry/openshift/e2e/test/qe/longrun/clusterinfra/openshift-e2e-test-qe-longrun-clusterinfra-chain.metadata.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+	"path": "openshift/e2e/test/qe/longrun/clusterinfra/openshift-e2e-test-qe-longrun-clusterinfra-chain.yaml",
+	"owners": {
+		"approvers": [
+			"jhou1",
+			"jianzhangbjz",
```
Comment (kuiwang02) [inline ci-operator/step-registry/openshift/e2e/test/qe/longrun/clusterinfra/openshift-e2e-test-qe-longrun-clusterinfra-chain.metadata.json:6]: @sunzhaohua2 please remove @jianzhangbjz and me.

File: ci-operator/step-registry/openshift-extended/test/clusterinfrastructure/openshift-extended-test-clusterinfrastructure-ref.yaml (hunk 1 above)
Comment (shellyyang1989) [reply ci-operator/step-registry/openshift-extended/test/clusterinfrastructure/openshift-extended-test-clusterinfrastructure-ref.yaml:3]: @sunzhaohua2 please add our team members as owner

Comment (sunzhaohua2) [reply ci-operator/step-registry/openshift-extended/test/clusterinfrastructure/openshift-extended-test-clusterinfrastructure-ref.yaml:3]: Thanks @kuiwang02 @shellyyang1989 do you think move this step to` step-registry/openshift/e2e/test/clusterinfra-qe` is ok? cc @liangxia

**Comment by sunzhaohua2** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-ibmcloud-ipi-private-fips-f28

File: ci-operator/step-registry/openshift/e2e/test/clusterinfra-qe/OWNERS [hunk 4]
```diff
@@ -0,0 +1 @@
+../OWNERS
```
Comment (liangxia) [inline ci-operator/step-registry/openshift/e2e/test/clusterinfra-qe/OWNERS:1]: Customize the owners to clusterinfra team members.

**Comment by sunzhaohua2** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-azure-upi-mini-perm-arm-f2

**Comment by sunzhaohua2** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-nutanix-ipi-data-disks-f28

**Comment by sunzhaohua2** [comment]
@liangxia please help to take a look, thanks!

File: ci-operator/step-registry/openshift/e2e/test/clusterinfra-qe/openshift-e2e-test-clusterinfra-qe-commands.sh [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+  failures: ${results[failures]}
+  errors: ${results[errors]}
+  skipped: ${results[skipped]}
+EOF
+    if [ ${results[failures]} != 0 ]; then
+        echo '  failingScenarios:' >>"${TEST_RESULT_FILE}"
+        # Every failure scenario will have a directory in machine-api-e2e-suite
+        readarray -t failingscenarios < <(find "${ARTIFACT_DIR}/machine-api-e2e-suite/" -mindepth 1 -maxdepth 1 -type d -exec basename {} \; | sort --unique)
+        for ((i = 0; i < ${#failingscenarios[@]}; i++)); do
+            echo "    - ${failingscenarios[$i]}" >>"${TEST_RESULT_FILE}"
```
Comment (liangxia) [inline ci-operator/step-registry/openshift/e2e/test/clusterinfra-qe/openshift-e2e-test-clusterinfra-qe-commands.sh:301]: One of the rehearse job failed to get the failingScenarios.

Comment (liangxia) [reply ci-operator/step-registry/openshift/e2e/test/clusterinfra-qe/openshift-e2e-test-clusterinfra-qe-commands.sh:301]: Not a blocker of this PR though. If you'd like to get this one in and use a follow up PR to fix that, I'm fine with that, too. Let me know which one do you prefer.

**Comment by sunzhaohua2** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-nutanix-ipi-data-disks-f28

File: ci-operator/step-registry/openshift/e2e/test/clusterinfra-qe/openshift-e2e-test-clusterinfra-qe-commands.sh (hunk 5 above)
Comment (sunzhaohua2) [reply ci-operator/step-registry/openshift/e2e/test/clusterinfra-qe/openshift-e2e-test-clusterinfra-qe-commands.sh]: @liangxia Done, please take a look when you got time.

**Pull request merged**